### PR TITLE
feat: integrate new deltalog format

### PR DESCRIPTION
**Pull request by tedxu**
See #34123

### CODE REVIEW

File: internal/datanode/importv2/task_l0_import_test.go [hunk 1]
```diff
@@ -191,7 +191,7 @@ func (s *L0ImportSuite) TestL0Import() {
 
 	deltaLog := actual.GetBinlogs()[0]
 	s.Equal(int64(s.delCnt), deltaLog.GetEntriesNum())
-	s.Equal(s.deleteData.Size(), deltaLog.GetMemorySize())
```
Comment (shaoting-huang) [inline internal/datanode/importv2/task_l0_import_test.go:194]: why is the assert removed

**Review by shaoting-huang — APPROVED** [review]

File: internal/datanode/importv2/task_l0_import_test.go (hunk 1 above)
Comment (tedxu) [reply internal/datanode/importv2/task_l0_import_test.go:194]: The serialized data sizes are not the same due to differing implementation. It is no longer a requirement that the two sizes here are equal.

File: internal/util/importutilv2/binlog/l0_reader.go [hunk 2]
```diff
@@ -82,24 +82,37 @@ func (r *l0Reader) Read() (*storage.DeleteData, error) {
...
+		defer reader.Close()
+
+		for {
+			err := reader.Next()
+			if err != nil {
+				if err == io.EOF {
+					break
 				}
-				deleteData.Append(dl.Pk, dl.Ts)
+				log.Error("compact wrong, fail to read deltalogs", zap.Error(err))
```
Comment (congqixia) [inline internal/util/importutilv2/binlog/l0_reader.go:108]: log content is misleading here, copies from compaction logic maybe?

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**